Examples/PSPDFCatalog/PSPDFCatalog/Catalog/Subclassing/CustomAnnotationProvider.cs

```diff
@@ -70,23 +72,44 @@ public CustomAnnotationProvider(PSPDFDocumentProvider documentProvider) : base(d
 					BoundingBox = new CGRect (100, pageInfo.Size.Height - 100, 32, 32),
 					Editable = false
 				};
-				if (!annotations.ContainsKey (pageIndex))
-					annotations.Add (pageIndex, new PSPDFAnnotation [] { noteAnnotation });
+				if (!annotationDict.ContainsKey (pageIndex))
+                    annotationDict.Add (pageIndex, new PSPDFAnnotation [] { noteAnnotation });
 			}
-			return annotations [pageIndex];
+			return annotationDict[pageIndex];
 		}
 
-		void PickColor (object o, EventArgs e)
+        public override PSPDFAnnotation[] AddAnnotations(PSPDFAnnotation[] annotations, NSDictionary<NSString, NSObject> options)
+        {
+            // Create an annotation list instead of an array because it's easier to work with
+            // We want to apply this to all annotation that are added (this only matters when you add 2 or more annotations at the same time via copy/paste etc)
+            foreach (var annotation in annotations)
+            {
+                // Need to add the already existing annotations
+                List<PSPDFAnnotation> existingAnnotations = annotationDict[annotation.PageIndex].ToList();
+                // Add all the other annotations once they're drawn
+                existingAnnotations.Add(annotation);
+                var annotationsAray = existingAnnotations.ToArray();
+                annotationDict[annotation.PageIndex] = annotationsAray;
```

Comment:
You could combine the above 2 lines.

```suggestion
                annotationDict[annotation.PageIndex] = existingAnnotations.ToArray();
```